backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/impl/BPartnerStatsDAO.java

```diff
@@ -338,4 +342,11 @@ private void updateCreditLimitIndicator(@NonNull final BPartnerStats bstats)
 
 		saveRecord(stats);
 	}
+
+	private void updateSectionCode(@NonNull final BPartnerStats bpStats)
+	{
+		final I_C_BPartner_Stats stats = loadDataRecord(bpStats);
+		stats.setM_SectionCode(stats.getM_SectionCode());
```

Comment:
Instead of resetting the section code to the same value, We should make sure we have the latest section code form the C_BPartner!

Reply:
TODO: 
Please, bring the service IBPartnerDAO, load the partner and copy the section code form it.